tests/unit/Store.ts

```diff
@@ -0,0 +1,261 @@
+import { EventObject, Hash } from 'dojo-core/interfaces';
+import * as lang from 'dojo-core/lang';
+import Promise from 'dojo-core/Promise';
+import * as dstore from 'src/interfaces'
+import registerSuite = require('intern!object');
+import assert = require('intern/chai!assert');
+import Store, { StoreArgs, NewableStoreModel } from 'src/Store';
+
+class NonDeclareBasedModel {
+	constructor (args?: {}) {
+		lang.mixin(this, args);
+	}
+	_restore(Constructor: { new (arg: any): { restored: boolean } }) {
+		// use constructor based restoration
+		const restored = new Constructor(this);
+		restored.restored = true;
+		return restored;
+	}
+	static createSubclass: any;
+}
+
+class Model {
+	constructor (args?: {}) {
+		lang.mixin(this, {
+			_restore: function (Constructor: { new (arg: any): { restored: boolean } }) {
```

Comment:
Why is this here when it is also below as a method?

For that matter, I'm wondering if we should just pull out the tests that test declare-specific stuff, since we don't expect declare to exist for Dojo 2.